joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/JavaSrc2Cpg.scala

```diff
@@ -47,4 +50,21 @@ class JavaSrc2Cpg {
     cpg
   }
 
+  /**
+    * JavaParser requires that the input path is a directory and not a single source file.
+    * This is inconvenient for small-scale testing, so if a single source file is created,
+    * copy it to a temp directory.
+    */
+  private def getSourcesFromDir(sourceCodePath: String): (String, List[String]) = {
+    val sourceFile = File(sourceCodePath)
+    if (sourceFile.isDirectory) {
+      val sourceFileNames = SourceFiles.determine(Set(sourceCodePath), sourceFileExtensions)
+      (sourceCodePath, sourceFileNames)
+    } else {
+      val dir = Files.createTempDirectory("javasrc")
```

Comment:
This temporary directory should be deleted / cleaned-up afterwards.

Reply:
Ah, good catch!